tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/HotChocolate/HotChocolateCommon.cs

```diff
@@ -0,0 +1,170 @@
+            where T : IQueryRequest
+        {
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            Scope scope = null;
+            try
+            {
+                var queryOperationName = request.OperationName;
+                var source = request.Query?.ToString();
+                var operationType = "Uncompleted";
+                scope = CreateScopeFromExecuteAsync(tracer, IntegrationId, new GraphQLTags(HotChocolateCommon.IntegrationName), ServiceName, queryOperationName, source, operationType);
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error creating or populating scope.");
+            }
+
+            return scope;
+        }
+
+        internal static void UpdateScopeFromExecuteAsync<T>(Tracer tracer, in T context)
+            where T : IOperationContext
+        {
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return;
+            }
+
+            var scope = tracer.InternalActiveScope;
+            var span = scope?.Span;
+            if (span == null || span.OperationName != ExecuteOperationName)
+            {
+                // not in a Hotchocolate execution span
+                return;
+            }
+
+            try
+            {
+                var operation = context.Operation;
+                var operationType = operation.OperationType.ToString();
+                if (span.Tags is GraphQLTags tags)
+                {
+                    span.ResourceName = $"{operationType} {tags.OperationName ?? "operation"}";
+                    tags.OperationType = operationType;
+                }
+                else
+                {
+                    var operationName = span.GetTag(Trace.Tags.GraphQLOperationName);
+                    span.ResourceName = $"{operationType} {operationName ?? "operation"}";
+                    span.SetTag(Trace.Tags.GraphQLOperationType, operationType);
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error updating HotChocolate scope.");
+            }
+        }
+
+        internal static void RecordExecutionErrorsIfPresent(Span span, string errorType, System.Collections.IEnumerable errors)
+        {
+            var executionErrors = GetList(errors);
+            var errorCount = executionErrors?.Count ?? 0;
+
+            if (errorCount > 0)
+            {
+                RecordExecutionErrors(span, errorType, errorCount, ConstructErrorMessage(executionErrors));
+            }
+        }
+
+        private static string ConstructErrorMessage(List<IError> executionErrors)
```

Comment:
NIT - I believe this method should have been in GraphQLCommon. Or at least the content of the for loop if easier.